include/bluetooth/radio_notification_cb.h

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#ifndef BT_RADIO_NOTIFICATION_CB_H__
+#define BT_RADIO_NOTIFICATION_CB_H__
+
+/**
+ * @file
+ * @defgroup bt_radio_notification_cb Radio Notification callback
+ * @{
+ * @brief APIs to set up radio notification callbacks
+ *
+ * The radio notification callbacks are triggered relative to the
+ * start of a Link Layer event.
+ *
+ * This information can be used to reduce the time from data sampling
+ * until data is sent on air.
+ */
+
+#include <stdint.h>
+#include <zephyr/bluetooth/conn.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** Recommended prepare distance for the connection event callback. */
+#define BT_RADIO_NOTIFICATION_CONN_CB_PREPARE_DISTANCE_US_RECOMMENDED 3000
```

Comment:
I think this is not enough if the application configures the maximum connection interval of 4s and using 500ppm clock accuracy.

Reply:
Maybe we should add some documentation to state that this value needs to be chosen according to the maximum connection interval and clock drift? We can calculate the maximum clock drift for some typically use cases

Reply:
not sure on what is best. We could consider
1. documenting until what connection interval the constant can be used
2. have different constants for different connection intervals (one for "short" and one for "long")
3. document on how the value can be chosen (e.g. provide some kind of formula)
4. have a runtime check based on the actual connection interval